blues/airnote.live
# feat: upgrade airnote svelte project to sveltekit

# Problem Context

There's big plans for the Airnote in the future, including expanded functionality like authentication and support for other products. Knowing that that's the direction it's headed, it's wise to take the project from Svelte to SvelteKit which offers us things like SSR, routing systems, and a host of other features that will make this future functionality easier to implement.

## Changes

Remake the app according to SvelteKit's specifications.

* Make it all TypeScript instead of JavaScript
* Use the Notehub JS library to interact with the Notehub API and fetch data from the Airnote project
* Use CSS variables for colors across the site
* Host the whole shebang on Netlify
* Port over all data fetching to `+page.server.ts` files
* Use SvelteKit file structure to define project routing
* Add `+layout.svelte` file to define project-level layout where header and footer stay the same and the page content itself is swapped in and out via the `<slot />`
* Move all random Svelte components, constants, icons, layout components, stores, utility functions, types and API service calls into the `lib` folder

## Screenshot (if applicable)

The UI should remain unchanged, but here's a deploy of the site running on Netlify, no backend server required any longer: https://airnote-live.netlify.app/dev:864475044215258/dashboard

## Testing

Unit / integration / e2e tests?

All unit and e2e tests manually. There's a future enhancement to add automatically running all tests in each PR to GitHub Action workflows.

Steps to test manually?

1. Go to https://airnote-live.netlify.app/dev:864475044215258/dashboard
2. Click around in the dashboard, the settings, see that the settings are not updatable without a valid PIN in the URL
3. Adjust things like C to F and refresh the page to make sure preferences are saved via local storage

Any other sorts of testing notes to include?

## Any other related PRs

n/a

## Ticket(s)

https://trello.com/c/vnqQEfrV/37-upgrade-airnote-project-to-sveltekit
